docs/controller.md

```diff
@@ -0,0 +1,108 @@
+# Architecture Document for Controller Component
+
+## Description
+The Controller is the component giving meaning to the [pipeline](#structure) file given by the Monitor. It receives [pipelines](#pipeline), parse them into a set of [actions](#actions) and send these actions sequentially to the Scheduler, for each of these actions, the Scheduler **must** notify the Controller when a job has been scheduled and has been completed successfully or encountered an error, it can be done just by sending the logs from the Agent for instance. Thanks to these information, the Scheduler is able to provide information about a pipeline state to anyone (the Monitor or any other client).
+## Features
+- Users can send pipelines containing actions to execute. These actions are basically shell commands to execute. These pipelines are `yaml` files.
+- Users can track there actions by getting the logs from the Agent, the states of the job : `TODO`, `PENDING`, `DOING`, `COMPLETED`, `ERROR`. Refer to the sections [actions/states](#States).
```

Comment:
Job or pipeline? Actions or pipelines? Do user submit a job, an action or a pipeline?

Users should not know that there are agents. They get their pipeline status from a controller HTTP endpoint. This endpoint may contain an actual log, but that will be defined by the corresponding openAPI definitions.

Reply:
The YAML file represents a pipeline, which can be defined as:
- the CI pipeline for the dev environment during a tag

This file will contain a list of actions (jobs have been standardised by action).
Examples of actions include Checking that the code is lint

The user will be able to retrieve the status of his pipeline with, for example, the status of all the actions that requested it, where the lint action is, where the build and push action is, where the chart helm update action is, etc...

The user will be able to send a YAML file (respecting the conventions requested) from a REST-API call.

Reply:
> The YAML file represents a pipeline, which can be defined as:
> 
>     * the CI pipeline for the dev environment during a tag

It's just a CI pipeline. Its relation to a development environment and a tag is only known to the end user.

> 
> This file will contain a list of actions (jobs have been standardised by action). Examples of actions include Checking that the code is lint
> 
> The user will be able to retrieve the status of his pipeline with, for example, the status of all the actions that requested it, where the lint action is, where the build and push action is, where the chart helm update action is, etc...
> 
> The user will be able to send a YAML file (respecting the conventions requested) from a REST-API call.

Reply:
We defined correctly pipelines and actions and we deleted the sentence about environment since it's not relevant